src/windows/models/properties_model.py

```diff
@@ -416,163 +419,152 @@ def value_updated(self, item, interpolation=-1, value=None, interpolation_detail
             % (property_key, clip_id, new_value, self.frame_number, interpolation, closest_point_x))
 
         # Find this clip
-        c = None
-        clip_updated = False
+        c = PropertiesModel.get_query_object(clip_id, item_type)
+        d = PropertiesModel.make_dataref(c, object_id)
+        if not c or property_key not in d:
+            log.error(
+                "%s %s: can't update %s property %s to %s",
+                item_type, clip_id, property_type, property_key, str(value))
+            self.parent.clearSelection()
+            return
 
-        if item_type == "clip":
-            # Get clip object
-            c = Clip.get(id=clip_id)
-        elif item_type == "transition":
-            # Get transition object
-            c = Transition.get(id=clip_id)
-        elif item_type == "effect":
-            # Get effect object
-            c = Effect.get(id=clip_id)
-
-        if c:
-            
-            # Create reference 
-            clip_data = c.data
-            if object_id:
-                clip_data = c.data.get('objects').get(object_id)
```

Comment:
Because of the `Effect.filter()` bug, whenever an effect's properties were being modified the reference made here (whether to the `Effect` object `data` member or the `data["objects"][object_id]` value it contained) allowed direct modification of the relevant `project.data['clips'][clip_id_pos]['effects'][effect_id_pos]` dict that held the property being changed.